index.js

```diff
@@ -2,26 +2,30 @@ var toDot = require('jsonpath-to-dot');
 
 module.exports = function(patches){
   var update = {};
+
   patches.map(function(p){
-    if(p.op === 'add'){
-      var path = toDot(p.path);
-      if(!update.$push) update.$push = {};
-      if(!update.$push[path]) {
-        update.$push[path] = p.value;
-      } else if (update.$push[path]) {
-        if(!update.$push[path].$each) {
-          update.$push[path] = {$each : [update.$push[path]]};
-        }
-        update.$push[path].$each.push(p.value);
-      }
+    if(p.op === 'add' || p.op === 'replace'){
+      if(!update.$set) update.$set = {};
+      update.$set[toDot(p.path)] = p.value;
     }
     else if(p.op === 'remove'){
-      if(!update.$unset) update.$unset = {};
+
+      var isArray = !p.hasOwnProperty('isArray') ? true : p.isArray;
```

Comment:
Where exactly is `isArray` coming from? The json-patch spec does not include it, does it?